aws/validators.go

```diff
@@ -903,6 +903,7 @@ func validateS3BucketLifecycleTransitionStorageClass() schema.SchemaValidateFunc
 		s3.TransitionStorageClassStandardIa,
 		s3.TransitionStorageClassOnezoneIa,
 		s3.TransitionStorageClassIntelligentTiering,
+		s3.StorageClassDeepArchive,
```

Comment:
Minor nit: Looks like there is an AWS Go SDK constant to match the others in this function: `TransitionStorageClassDeepArchive` -- will adjust this on merge so we can release this today.